addons/account/wizard/account_move_send.py

```diff
@@ -245,16 +245,9 @@ def _compute_send_mail_extra_fields(self):
         for wizard in self:
             wizard.enable_send_mail = wizard.mode in ('invoice_single', 'invoice_multi')
             wizard.display_mail_composer = wizard.mode == 'invoice_single'
-            wizard.send_mail_warning_message = False
-
             invoices_without_mail_data = wizard.move_ids.filtered(lambda x: not x.partner_id.email)
             wizard.send_mail_readonly = invoices_without_mail_data == wizard.move_ids
-
-            if wizard.mode == 'invoice_multi' and wizard.checkbox_send_mail and invoices_without_mail_data:
-                wizard.send_mail_warning_message = _(
-                    "The partners on the following invoices have no email address, "
-                    "so those invoices will not be sent: %s",
-                    ", ".join(invoices_without_mail_data.mapped('name')))
+            wizard.send_mail_warning_message = 'missing_info' if invoices_without_mail_data else False
```

Comment:
Maybe it's worth changing the send_mail_warning_message to a fields.Selection then ?